### ISSUE TITLE
[bug] CMakeToolchain doesn't work with NDK r18b

### ISSUE THREAD

**Issue by DDoSolitary** [conan-io/conan#11798]
### Environment Details (include every applicable attribute)
  * Operating System+version: host: Ubuntu 18.04, target: Android
  * Compiler+version: Clang 7.0 (NDK r18b)
  * Conan version: 1.51.0
  * Python version: 3.8.0

### Steps to reproduce (Include if Applicable)

```bash
conan install --build=missing -pr ndk.profile -pr:b default -s build_type=Debug -c tools.android:ndk_path=${ANDROID_NDK_ROOT}
```

ndk.profile:
```ini
[settings]
os=Android
os_target=Android
os_build=Linux
arch=armv7
arch_target=armv7
arch_build=x86_64
compiler=clang
compiler.version=7.0
compiler.libcxx=c++_shared
os.api_level=21
[options]
[env]
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
Configuration (profile_host):
[settings]
arch=armv7
arch_build=x86_64
arch_target=armv7
build_type=Debug
compiler=clang
compiler.libcxx=c++_shared
compiler.version=7.0
os=Android
os.api_level=21
os_build=Linux
os_target=Android
[options]
[build_requires]
[env]
[conf]
tools.android:ndk_path=/opt/android-ndk-r18b

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=7
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

Installing package: zlib-ng/2.0.6
Requirements
    zlib-ng/2.0.6 from 'conancenter' - Cache
Packages
    zlib-ng/2.0.6:bc0ad90edd3c2036309c6af1cb019698f094f6aa - Build

Cross-build from 'Linux:x86_64' to 'Android:armv7'
Installing (downloading, building) binaries...
zlib-ng/2.0.6: Copying sources to build folder
zlib-ng/2.0.6: Building your package in /home/<redacted>/.conan/data/zlib-ng/2.0.6/_/_/build/bc0ad90edd3c2036309c6af1cb019698f094f6aa
zlib-ng/2.0.6: Generator txt created conanbuildinfo.txt
zlib-ng/2.0.6: Calling generate()
zlib-ng/2.0.6: Aggregating env generators
zlib-ng/2.0.6: Calling build()
zlib-ng/2.0.6: CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/<redacted>/.conan/data/zlib-ng/2.0.6/_/_/build/bc0ad90edd3c2036309c6af1cb019698f094f6aa/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/<redacted>/.conan/data/zlib-ng/2.0.6/_/_/package/bc0ad90edd3c2036309c6af1cb019698f094f6aa" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Debug" "/home/<redacted>/.conan/data/zlib-ng/2.0.6/_/_/build/bc0ad90edd3c2036309c6af1cb019698f094f6aa/source_subfolder"
-- Using CMake version 3.16.9
-- ZLIB_HEADER_VERSION: 1.2.11
-- ZLIBNG_HEADER_VERSION: 2.0.6
-- Using Conan toolchain: /home/<redacted>/.conan/data/zlib-ng/2.0.6/_/_/build/bc0ad90edd3c2036309c6af1cb019698f094f6aa/conan_toolchain.cmake
CMake Error at /opt/android-ndk-r18b/build/cmake/android.toolchain.cmake:513 (message):
  Invalid Android platform: 21.
Call Stack (most recent call first):
  ../conan_toolchain.cmake:26 (include)
  /usr/local/share/cmake-3.16/Modules/CMakeDetermineSystem.cmake:93 (include)
  CMakeLists.txt:37 (project)


CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
zlib-ng/2.0.6:
zlib-ng/2.0.6: ERROR: Package 'bc0ad90edd3c2036309c6af1cb019698f094f6aa' build failed
zlib-ng/2.0.6: WARN: Build folder /home/<redacted>/.conan/data/zlib-ng/2.0.6/_/_/build/bc0ad90edd3c2036309c6af1cb019698f094f6aa
ERROR: zlib-ng/2.0.6: Error in build() method, line 68
        cmake.configure(build_script_folder=self._source_subfolder)
        ConanException: Error 1 while executing cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/<redacted>/.conan/data/zlib-ng/2.0.6/_/_/build/bc0ad90edd3c2036309c6af1cb019698f094f6aa/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/<redacted>/.conan/data/zlib-ng/2.0.6/_/_/package/bc0ad90edd3c2036309c6af1cb019698f094f6aa" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Debug" "/home/<redacted>/.conan/data/zlib-ng/2.0.6/_/_/build/bc0ad90edd3c2036309c6af1cb019698f094f6aa/source_subfolder"
```

**Comment by DDoSolitary** [comment]
CMakeToolchain directly sets ANDROID_PLATFORM to the API level number. While this is supported by recent NDK versions, older ones require the `android-<level>` format. NDK r18b is one of those old versions and unfortunately is the version that my company is still using.

**Comment by franramirez688** [comment]
Thanks for the contribution! The fix will be released in the next Conan 1.52 version.
